modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -1093,11 +1094,35 @@ CompletableFuture<Void> saveDataNodesToMetaStorageOnScaleUp(int zoneId, long rev
                     return completedFuture(null);
                 }
 
-                Map<String, Integer> dataNodesFromMetaStorage = fromBytes(values.get(zoneDataNodesKey(zoneId)).value());
+                Entry dataNodesValue = values.get(zoneDataNodesKey(zoneId));
```

Comment:
What we expect to retrieve from the meta storage if there are no corresponding keys?
We will get map with three elements, right? 
The values will be null? or empty? See Entry#empty?
Is it possible to have null/empty dataNodesValue and non null/empty zoneScaleUpChangeTriggerValue?

Reply:
Fixed. We can use Entry#empty.